pypdf/_codecs/_codecs.py

```diff
@@ -138,8 +139,123 @@ def _pack_codes_into_bytes(self, codes: List[int]) -> bytes:
 
         return bytes(output)
 
-    def decode(self, data: bytes) -> bytes:
-        """Decode data using LZW."""
-        from ..filters import LZWDecode
+    def _next_code(self, data: bytes) -> int:
+        self.bitpos: int
+        self._next_bits: int
+
+        fillbits = self.bits_per_code
+        value = 0
+
+        while fillbits > 0:
+            if self._byte_pointer >= len(data):
+                return -1
+
+            nextbits = data[self._byte_pointer]
+            bits_available = 8 - self.bitpos
+            bits_to_use = min(bits_available, fillbits)
+
+            value |= (
+                (nextbits >> (8 - self.bitpos - bits_to_use))
+                & (0xFF >> (8 - bits_to_use))
+            ) << (fillbits - bits_to_use)
+
+            fillbits -= bits_to_use
+            self.bitpos += bits_to_use
+
+            if self.bitpos == 8:
+                self.bitpos = 0
+                self._byte_pointer += 1
+
+        return value
 
-        return LZWDecode.Decoder(data).decode()
+    def next_code_decode(self) -> int:
+        self._next_data: int
+        assert self._data
+        try:
+            while self._next_bits < self._bits_to_get:
+                self._next_data = (self._next_data << 8) | (
+                    self._data[self._byte_pointer] & 0xFF
+                )
+                self._byte_pointer += 1
+                self._next_bits += 8
+
+            code = (
+                self._next_data >> (self._next_bits - self._bits_to_get)
+            ) & self._and_table[self._bits_to_get - 9]
+            self._next_bits -= self._bits_to_get
+
+            return code
+        except IndexError:
+            return 257  # End of data
+
+    def decode(self, data: bytes) -> bytes:
+        """
+        The following code was converted to Python from the following code:
+        https://github.com/empira/PDFsharp/blob/master/src/foundation/src/PDFsharp/src/PdfSharp/Pdf.Filters/LzwDecode.cs
+        """
+        self._and_table = [511, 1023, 2047, 4095]
+        self._string_table: list[bytes]
+        self._data = None
+        self._table_index = 0
+        self._bits_to_get = 9
+        self._byte_pointer = 0
+        self._next_data = 0
+        self._next_bits = 0
+
+        if data[0] == 0x00 and data[1] == 0x01:
+            raise Exception("LZW flavor not supported.")
```

Comment:
This should use a more specific exception.

Reply:
I've removed the check completely